### PR TITLE
Allow `indexing_threshold=null` to disable indexing

### LINKED ISSUE #1621: Allow `indexing_threshold_kb: null` in config to disable indexing
**Is your feature request related to a problem? Please describe.**
Currently, it is recommended to set some large value to disable indexing, but it looks ugly.

**Describe the solution you'd like**
Allow `indexing_threshold_kb: null`, which should be considered as disabled indexing

### PR DESCRIPTION
**Pull request by coszio**
The suggested way of disabling indexing is to put a really high number in this field, so I just converted it to an Option, and in case it is None -> it becomes usize::max.

Fixes #1621 

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally using ``cargo fmt`` command prior to submission?
3. [x] Have you checked your code using ```cargo clippy``` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by agourlay** [comment]
Great to see the schema consistency check kicking in (cc @timvisee)

**Comment by agourlay** [comment]
One idea for testing could be using the [collection-info](https://qdrant.tech/documentation/collections/#collection-info) to retrieve the number of indexed vectors

**Review by timvisee — APPROVED** [review]
LGTM!

File: openapi/tests/openapi_integration/test_vector_config.py [hunk 1]
```diff
@@ -142,3 +143,92 @@ def test_retrieve_vector_specific_quantization():
     assert vectors['text']['quantization_config']['scalar']['always_ram']
     assert config['quantization_config']['scalar']['type'] == "int8"
     assert config['quantization_config']['scalar']['quantile'] == 0.5
+
+@pytest.mark.skip(reason="Takes too long for a sanity test")
+def test_disable_indexing():
+    indexed_name = 'test_collection_indexed'
+    unindexed_name = 'test_collection_unindexed'
+    
+    response = request_with_validation(
```
Comment (timvisee) [inline openapi/tests/openapi_integration/test_vector_config.py:152]: On second thought; this should probably be done at the end of the test to properly clean up, asserting can be skipped.

The `drop_collection` function can be used as helper for this: https://github.com/qdrant/qdrant/blob/4b03cf66ab365b82b3a76212d7fb10809b43317d/openapi/tests/openapi_integration/test_vector_config.py#L13

**Review by generall — APPROVED** [review]

**Review by agourlay — APPROVED** [review]

**Pull request merged**